netmap/attributes.go

```diff
@@ -0,0 +1,208 @@
+package netmap
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/nspcc-dev/neofs-api-go/v2/refs"
+)
+
+// prefix of keys to subnet attributes.
+const attrSubnetPrefix = "__NEOFS__SUBNET_"
+
+const (
+	// subnet attribute's value denoting subnet entry
+	attrSubnetValEntry = "True"
+
+	// subnet attribute's value denoting subnet exit
+	attrSubnetValExit = "False"
+)
+
+// NodeSubnetInfo groups information about subnet which can be written to NodeInfo.
+//
+// Zero value represents entry to zero subnet.
+type NodeSubnetInfo struct {
+	exit bool
+
+	id *refs.SubnetID
+}
+
+// Enters returns true iff node enters the subnet.
+func (x NodeSubnetInfo) Enters() bool {
+	return !x.exit
+}
+
+// SetEntryFlag sets the subnet entry flag.
+func (x *NodeSubnetInfo) SetEntryFlag(enters bool) {
+	x.exit = !enters
+}
+
+// ID returns identifier of the subnet.
+func (x NodeSubnetInfo) ID() *refs.SubnetID {
+	return x.id
+}
+
+// SetID sets identifier of the subnet.
+func (x *NodeSubnetInfo) SetID(id *refs.SubnetID) {
+	x.id = id
+}
+
+func subnetAttributeKey(id *refs.SubnetID) string {
+	txt, _ := id.MarshalText() // never returns an error
+
+	return attrSubnetPrefix + string(txt)
+}
+
+// WriteSubnetInfo writes NodeSubnetInfo to NodeInfo via attributes. NodeInfo must not be nil.
+//
+// Does not add (removes existing) attribute if node:
+//   * exists non-zero subnet;
+//   * enters zero subnet.
+//
+// Attribute key is calculated from ID using format `__NEOFS__SUBNET_%s`.
+// Attribute Value is:
+//   * `True` if node enters the subnet;
+//   * `False`, otherwise.
+func WriteSubnetInfo(node *NodeInfo, info NodeSubnetInfo) {
+	attrs := node.GetAttributes()
+
+	id := info.ID()
+	enters := info.Enters()
+
+	// calculate attribute key
+	key := subnetAttributeKey(id)
+
+	if refs.IsZeroSubnet(id) == enters {
+		for i := range attrs {
+			if attrs[i].GetKey() == key {
+				attrs = append(attrs[:i], attrs[i+1:]...)
+			}
+		}
+	} else {
+		var val string
+
+		if enters {
+			val = attrSubnetValEntry
+		} else {
+			val = attrSubnetValExit
+		}
+
+		presented := false
+
+		for i := range attrs {
+			if attrs[i].GetKey() == key {
+				attrs[i].SetValue(val)
+				presented = true
+			}
+		}
+
+		if !presented {
+			var attr Attribute
+
+			attr.SetKey(key)
+			attr.SetValue(val)
+
+			attrs = append(attrs, &attr)
+		}
+	}
+
+	node.SetAttributes(attrs)
+}
+
+// ErrRemoveSubnet is returned when a node needs to leave the subnet.
+var ErrRemoveSubnet = errors.New("remove subnet")
+
+// IterateSubnets iterates over all subnets the node belongs to and passes the IDs to f.
+// Handler must not be nil.
+//
+// If f returns ErrRemoveSubnet, then removes subnet entry. Breaks on any other non-nil error and returns it.
+//
+// Returns an error if any subnet attribute has wrong format.
```

Comment:
Placement policy currently checks only `False` value for the default policy and `True` for the others. All non-default values are considered equal. So we should either remove the attributes from the node or to agree to behave similar to the policy. Removing the node from netmap is also an option.

Reply:
API protocol allows only `False` and `True` values, so we should consider any other values incorrect.

Reply:
currently yes, API declares exactly two values and says that they are the only correct opportunities 

I think any other values should be considered incorrect to prevent `true` and `false` confusing values (the first letter is not capitalized) OR we should read such attributes case-insensitively (and, therefore, change API desc)